interfaces/inc/c_pal/malloc_multi_flex.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+#ifndef MALLOC_MULTI_FLEX_H
+#define MALLOC_MULTI_FLEX_H
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+    #define ARG_OVERFLOW_CHECK(arg) \
+        if ((MU_C2(arg, _size) != 0 && SIZE_MAX / MU_C2(arg, _size) < MU_C2(arg, _count)) || (SIZE_MAX - size_tracker < MU_C2(arg, _count) * MU_C2(arg, _size)))\
+        {\
+            return NULL;\
+        }\
+        size_tracker += MU_C2(arg, _count) * MU_C2(arg, _size);
+
+    #define ARGS_OVERFLOW_CHECK(...) \
+        size_t size_tracker = parent_struct_size;\
```

Comment:
>size_t size_tracker = parent_struct_size;\ [](http://example.com/codeflow?start=8&length=42)

slightly more readable IMO if this line is in the main function like around line 49-50